kong/plugins/ldap-auth/access.lua

```diff
@@ -165,7 +165,14 @@ local function do_authentication(conf)
 
   -- If both headers are missing, return 401
   if not (authorization_value or proxy_authorization_value) then
-    ngx.header["WWW-Authenticate"] = 'LDAP realm="kong"'
+    local scheme = conf.header_type
+    if scheme == "ldap" then
+      -- ensure backwards compatibility (see GH PR #3656)
+      -- TODO: provide migration to capitalize older configurations
+      scheme = upper(scheme)
```

Comment:
`upper` isn't a valid Lua function, `string.upper` is. I was suggesting caching it at the top of this module, like `string.lower`

Reply:
yep... sorry, did it in a rush before leaving the office.
The plugin unit tests did pass, though the [cassandra integration job](https://travis-ci.org/Kong/kong/jobs/411478395) failed with an error that does not seem related to by patch:
```
[  FAILED  ] spec/02-integration/02-cmd/02-start_stop_spec.lua @ 114: kong start/stop /etc/hosts resolving in CLI resolves #cassandra hostname
```

Reply:
@francois-maillard Spurious failures can happen in the CI, in particular on PR builds. It is something we are working towards fixing, no worries for now, I'll restart the jobs if those failures are unrelated.

Reply:
All tests are green!